src/braket/default_simulator/openqasm/circuit.py

```diff
@@ -63,28 +64,47 @@ def num_qubits(self) -> int:
 
     @property
     def basis_rotation_instructions(self):
-        """
-        This function assumes all observables are commuting.
-        """
-
         basis_rotation_instructions = []
-        measured_qubits = set()
+        observable_map = {}
+
+        def process_observable(observable):
+            if isinstance(observable, Identity):
+                return
+            measured_qubits = tuple(observable.measured_qubits)
+            for qubit in measured_qubits:
+                for target, previously_measured in observable_map.items():
+                    if qubit in target:
+                        conflicts = False
+                        if not (
+                            target == measured_qubits
+                            and type(previously_measured) == type(observable)
+                        ):
+                            conflicts = True
+                        elif (
+                            isinstance(observable, Hermitian)
+                            and isinstance(previously_measured, Hermitian)
+                            and not np.allclose(observable.matrix, previously_measured.matrix)
```

Comment:
Also need to ensure that the target qubits are in the same order for a multi-qubit observable. For example, a 2-qubit observable is generally not simultaneously measurable on targets [0. 1] and [1, 0].

Reply:
My code should be doing that now. The targets are being checked as tuples, so order counts

Reply:
D'oh, somehow simultaneously missed that for this comment and yet caught it for the next one